x-pack/plugins/gis/public/shared/layers/tile_layer.js

```diff
@@ -32,14 +33,64 @@ export class TileLayer extends ALayer {
     return tileLayerDescriptor;
   }
 
+  _tileLoadErrorTracker(map, url) {
+    let tileLoad;
+    map.on('dataloading', ({ tile }) => {
+      if (tile && tile.request) {
+        // If at least one tile loads, endpoint/resource is valid
+        tile.request.onloadend = ({ loaded }) => loaded && (tileLoad = true);
+      }
+    });
 
-  syncLayerWithMB(mbMap) {
+    return new Promise((resolve, reject) => {
+      let tileLoadTimer = null;
+      let checkInterval = null;
 
+      const clearChecks = () => {
+        clearInterval(checkInterval);
+        clearTimeout(tileLoadTimer);
+        map.off('dataloading');
+      };
+
+      checkInterval = setInterval(() => {
```

Comment:
Why is this interval needed? Isn't the timer sufficient?

Reply:
Running timers stress me out :), especially ones that run for 32 seconds as this one does. I added this on a second pass to clear the timer as soon as the tiles have loaded rather than keep running. I don't mind removing it though. It's a little cluttery and we can always bring it back if we want.